### ISSUE TITLE
Add KNN into `Trainer` for use in validating SSL models

### ISSUE THREAD

**Issue by Pale-Blue-Dot-97** [Pale-Blue-Dot-97/Minerva#140]
## Issue

Validation loss and accuracy by using partitioned data from the training dataset is not a reliable way of validating a SSL model's performance during fitting. A much better approach is training a KNN using the backbone from the SSL model on validation data. 

## Solution

~- [ ] Add a KNN model class into `models`.~
~- [ ] Add checks in `Trainer.__init__` for SSL models and activate KNN validation.~
- [x] Add new form of `epoch` method to `Trainer` to perform KNN validation.
- [x] Reorder `Trainer.fit` to only run a validation epoch at more intermittent points (typically every 5 epochs of training).
- [x] Add new test in `test_trainer.py` to cover a SSL model training using KNN validation to close most remaining coverage gap.
